src/main/scala/nl.knaw.dans.easy.download/components/AuthenticationComponent.scala

```diff
@@ -69,6 +69,8 @@ trait AuthenticationComponent extends DebugEnhancedLogging {
           .tried
       }.recoverWith {
         case t: AuthenticationException => Failure(InvalidUserPasswordException(userName, new Exception("invalid password", t)))
+        case t: InvalidNameException => Failure(InvalidNameAuthenticationException(t))
```

Comment:
No need for three different exceptions. Use something like 

    case t: InvalidNameException | InvalidNameException => Failure(InvalidUserPasswordException(...))

It also reduces the pattern match in `EasyDownloadServlet`

Reply:
When you use `InvalidUserPasswordException`, you will not see the `cause` of the failure.

In InvalidUserPasswordException:
      `logger.info(s"invalid credentials for $userName: ${ cause.getMessage }")`

In the case of `InvalidNameException` and `OperationNotSupportedException` it is better to log the `cause` of the exception.

Reply:
By the way, the error message says "invalid password". 

Response to the end user should never tell whether the user name or password was invalid, just that the combination is invalid. An invalid password might tell a hacker he guessed the user right. Internally that difference might be important whenever we decide to add a policy to increase response time for a user on an invalid password. Though I rather think that should be an ldap feature than an application feature.

Reply:
The user gets in these situations also only a message that he is unauthorized.  The more detailed cause comes only in the log file.